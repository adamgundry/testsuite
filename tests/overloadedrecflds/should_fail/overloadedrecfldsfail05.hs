{-# LANGUAGE OverloadedRecordFields, TypeFamilies, FlexibleInstances,
             DataKinds, MultiParamTypeClasses #-}

import GHC.Records

data Person = MkPerson { firstName :: String, lastName :: String }

type instance GetResult Person "fullName" = String
instance Has Person "fullName" String where
  getField _ p = firstName p ++ " " ++ lastName p
