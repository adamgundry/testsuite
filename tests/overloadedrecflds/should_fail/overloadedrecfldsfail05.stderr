
overloadedrecfldsfail05.hs:8:15:
    Illegal type instance declaration for ‛GetResult’
      (Use -XOverloadedRecordFields instead.)
    In the type instance declaration for ‛GetResult’

overloadedrecfldsfail05.hs:9:10:
    Illegal instance declaration for ‛Has Person "fullName" String’
      (Use -XOverloadedRecordFields instead.)
    In the instance declaration for ‛Has Person "fullName" String’
